### PR TITLE
Implement `SearchHeaderViewController` and `SearchHeaderViewModel`

### PR DESCRIPTION
**Pull request by lionel-alves**
Depends on https://github.com/lionel-alves/TranslateKit/pull/11.

### CODE REVIEW

File: weTranslate/SearchHeaderViewController.swift [hunk 1]
```diff
@@ -32,11 +48,74 @@ final class SearchHeaderViewController: UIViewController {
...
+        isExpended = !isExpended
+        self.searchHeaderView.languagesPickerView.alpha = self.isExpended ? 1 : 0
+        animate {
+            self.searchHeaderView.languagesPickerView.hidden = !self.isExpended
+        }
+    }
+}
+
+
+
```
Comment (lionel-alves) [inline weTranslate/SearchHeaderViewController.swift]: :scissors:

File: weTranslate/SearchHeaderViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    var selectedRow: Int {
+        let language = fromLanguage == .English ? toLanguage : fromLanguage
+        return languages.indexOf(language) ?? 0
+    }
+
+    private var fromEnglish: Bool = true
+
+    // MARK: - Properties
```
Comment (lionel-alves) [inline weTranslate/SearchHeaderViewModel.swift]: `Properties` -> `Methods`

File: weTranslate/SearchHeaderViewController.swift [hunk 3]
```diff
@@ -32,11 +48,74 @@ final class SearchHeaderViewController: UIViewController {
...
+        searchHeaderView.fromLanguageButton.addTarget(self, action: "selectLanguage:", forControlEvents: .TouchUpInside)
+        searchHeaderView.toLanguageButton.addTarget(self, action: "selectLanguage:", forControlEvents: .TouchUpInside)
+        searchHeaderView.swapButton.addTarget(self, action: "swap:", forControlEvents: .TouchUpInside)
+
+        view.addSubview(searchHeaderView)
+
+        searchHeaderView.leadingAnchor.constraintEqualToAnchor(view.leadingAnchor).active = true
+        searchHeaderView.trailingAnchor.constraintEqualToAnchor(view.trailingAnchor).active = true
+        searchHeaderView.topAnchor.constraintEqualToAnchor(view.topAnchor).active = true
+        searchHeaderView.bottomAnchor.constraintEqualToAnchor(view.bottomAnchor).active = true
```
Comment (jenniferlim) [inline weTranslate/SearchHeaderViewController.swift:65]: what about creating a func that init constraints

Comment (lionel-alves) [inline weTranslate/SearchHeaderViewController.swift:65]: See https://github.com/jenniferlim/weTranslate/issues/9, will do that in a separate PR.

File: weTranslate/SearchHeaderViewController.swift [hunk 4]
```diff
@@ -32,11 +48,74 @@ final class SearchHeaderViewController: UIViewController {
...
+    func swap(sender: UIButton?) {
+        viewModel.swap()
+    }
+
+    func selectLanguage(sender: UIButton?) {
+        isExpended = !isExpended
+        self.searchHeaderView.languagesPickerView.alpha = self.isExpended ? 1 : 0
+        animate {
+            self.searchHeaderView.languagesPickerView.hidden = !self.isExpended
+        }
```
Comment (jenniferlim) [inline weTranslate/SearchHeaderViewController.swift:88]: would it be better to create a private func : updatePickerUI based on isExpended ?

File: weTranslate/SearchHeaderViewController.swift [hunk 5]
```diff
@@ -32,11 +48,74 @@ final class SearchHeaderViewController: UIViewController {
...
+    }
+}
+
+
```
Comment (jenniferlim) [inline weTranslate/SearchHeaderViewController.swift:92]: :scissors:

File: weTranslate/SearchHeaderViewController.swift [hunk 6]
```diff
@@ -32,11 +48,74 @@ final class SearchHeaderViewController: UIViewController {
...
+    func pickerView(pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        return viewModel.languages.count
+    }
+}
+
+
+extension SearchHeaderViewController: UIPickerViewDelegate {
+
+    func pickerView(pickerView: UIPickerView, attributedTitleForRow row: Int, forComponent component: Int) -> NSAttributedString? {
+        let attributes = [NSForegroundColorAttributeName : UIColor.whiteColor()]
```
Comment (jenniferlim) [inline weTranslate/SearchHeaderViewController.swift:120]: .whiteColor() instead of UIColor.whiteColor()

Comment (lionel-alves) [inline weTranslate/SearchHeaderViewController.swift:120]: :+1:

File: weTranslate/SearchHeaderViewModel.swift [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+//  Copyright © 2016 weTranslate. All rights reserved.
+//
+
+import Foundation
+import TranslateKit
+
+struct SearchHeaderViewModel {
+
+    // MARK: - Properties
+
```
Comment (jenniferlim) [inline weTranslate/SearchHeaderViewModel.swift:15]: Should we add a newline here ?

File: weTranslate/SearchHeaderViewModel.swift [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    var selectedRow: Int {
+        let language = fromLanguage == .English ? toLanguage : fromLanguage
+        return languages.indexOf(language) ?? 0
+    }
+
+    private var fromEnglish: Bool = true
+
+    // MARK: - Properties
+
```
Comment (jenniferlim) [inline weTranslate/SearchHeaderViewModel.swift]: should we add a newline here ?
